app/code/Magento/Inventory/Model/SourceItem/Validator/SourceItemsValidator.php

```diff
@@ -0,0 +1,57 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\Inventory\Model\SourceItem\Validator;
+
+use Magento\Framework\Validation\ValidationResult;
+use Magento\Framework\Validation\ValidationResultFactory;
+use Magento\InventoryApi\Api\Data\SourceItemInterface;
+
+/**
+ * Responsible for Source items validation
+ */
+class SourceItemsValidator
+{
+    /**
+     * @var SourceItemValidatorInterface
+     */
+    private $sourceItemValidator;
+
+    /**
+     * @var ValidationResultFactory
+     */
+    private $validationResultFactory;
+
+    /**
+     * @param ValidationResultFactory $validationResultFactory
+     * @param SourceItemValidatorInterface $sourceItemValidator
+     */
+    public function __construct(
+        ValidationResultFactory $validationResultFactory,
+        SourceItemValidatorInterface $sourceItemValidator
+    ) {
+        $this->validationResultFactory = $validationResultFactory;
+        $this->sourceItemValidator = $sourceItemValidator;
+    }
+
+    /**
+     * @param SourceItemInterface[] $sourceItems
+     * @return ValidationResult
+     */
+    public function validate(array $sourceItems): ValidationResult
+    {
+        $errors = [[]];
+        foreach ($sourceItems as $sourceItem) {
+            $validationResult = $this->sourceItemValidator->validate($sourceItem);
```

Comment:
This is not really a defensive style of programming, as array types are not checked anywhere before this, and Strict Types error may be rised by the interpreter